apps/nextjs-website/src/lib/__tests__/header.test.ts

```diff
@@ -0,0 +1,139 @@
+import { describe, it, expect } from 'vitest';
+import { makeHeaderProps } from '../header'; // Adjust the import path based on your file structure
+import { Header } from '../fetch/header';
+import { CTAButtonType } from '../fetch/types/CTAButton';
+
+const parentNavItem = {
+  order: 1,
+  id: 1,
+  title: 'Parent',
+  path: 'parent',
+  parent: null,
+  menuAttached: true,
+};
+
+const childNavItem = {
+  order: 1,
+  id: 2,
+  title: 'Child',
+  path: 'child',
+  parent: parentNavItem,
+  menuAttached: true,
+};
+
+const navigationWithParentAndChild = [parentNavItem, childNavItem];
+
+const productName = 'My Product';
+const ctaButtons: ReadonlyArray<CTAButtonType> = [
+  {
+    text: 'Primary',
+    href: 'primary.com',
+    variant: 'contained',
+    color: 'inherit',
+  },
+  {
+    text: 'Secondary',
+    href: 'secondary.com',
+    variant: 'outlined',
+    color: 'inherit',
+  },
+];
+
+const header: Header = {
+  data: {
+    attributes: {
+      theme: 'light',
+      avatar: {
+        data: {
+          attributes: {
+            url: 'path/to/avatar.jpg',
+            alternativeText: 'Alt text', // Add this line if alternativeText is part of your data structure
```

Comment:
```suggestion
            alternativeText: 'Alt text',
```
is it a part of the structure or not? 🤔
do we want it to be tested or not?